app/tests/types/i18next.test.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @jest-environment node
+ */
+import fs from "fs";
+import tsEnglishResources from "src/types/generated-i18n-bundle";
+
+/**
+ * Add a custom matcher so we can provide a more helpful test failure message
+ */
+function toHaveI18nNamespaces(received: string[], expected: string[]) {
+  const missingNamespaces = expected.filter(
+    (namespace) => !received.includes(namespace)
+  );
+
+  return {
+    pass: missingNamespaces.length === 0,
+    message: () => {
+      const missingNamespacesString = missingNamespaces.join(", ");
+      let message = `The src/types/generated-i18n-bundle.ts file is missing imports for these English namespaces: ${missingNamespacesString}`;
+      message += `\n\nYou can fix this by re-running "npm run i18n-types" to regenerate the file.`;
+      message += `\n\nYou likely added a new namespace to the English locale files but the i18n-types script hasn't ran yet.`;
+      message += `\n\nIt's important for the generated-i18n-bundle.ts file to be up to date so that you don't get inaccurate TypeScript errors.`;
+
+      return message;
+    },
+  };
+}
+
+expect.extend({ toHaveI18nNamespaces });
+
+describe("types/generated-i18n-bundle.ts", () => {
+  it("includes all English namespaces", () => {
+    const jsonEnglishFilenames = fs
```

Comment:
is each file supposed to be a namespace? if so, i think the variable name (and the assertion below) would read more clearly if it was called namespaces e.g. 
```suggestion
    const i18nNamespaces = fs
```

Reply:
Yep, I believe `filename == namespace` in i18next